packages/modules/devices/e3dc/device.py

```diff
@@ -102,7 +102,7 @@ def read_legacy_bat(address1: str,
             soc_tmp, power_tmp = read_bat(client)
             soc += soc_tmp
             power += power_tmp
-            pv_tmp, pv_external_tmp = read_inverter(client, read_ext)
+            pv_tmp, pv_external_tmp = read_inverter(client, cast(bool, read_ext))
```

Comment:
```suggestion
            pv_tmp, pv_external_tmp = read_inverter(client, bool(read_ext))
```
Wie Yannik [hier](https://github.com/snaptec/openWB/pull/2496#discussion_r1024489395) erklärt hat, sagt cast, dass `read_ext` ein Bool ist, read_ext ist aber weiterhin ein int. Entsprechend auch an den anderen Stellen, an denen cast zum Einsatz kommt.

Reply:
Bool am Anfang gesetzt